brave/src/main/java/com/linecorp/armeria/server/brave/BraveService.java

```diff
@@ -23,80 +23,73 @@
 import com.linecorp.armeria.server.SimpleDecoratingService;
 
 import brave.Span;
-import brave.Span.Kind;
 import brave.Tracer;
 import brave.Tracer.SpanInScope;
-import brave.Tracing;
+import brave.http.HttpServerHandler;
+import brave.http.HttpTracing;
 import brave.propagation.TraceContext;
-import brave.propagation.TraceContextOrSamplingFlags;
 
 /**
  * Decorates a {@link Service} to trace inbound {@link HttpRequest}s using
  * <a href="https://github.com/openzipkin/brave">Brave</a>.
  */
 public final class BraveService extends SimpleDecoratingService<HttpRequest, HttpResponse> {
-
-    // TODO(minwoox) Add the variant which takes HttpTracing.
-
     /**
-     * Creates a new tracing {@link Service} decorator using the specified {@link Tracing} instance.
+     * Creates a new tracing {@link Service} decorator using the specified {@link HttpTracing} instance.
      */
     public static Function<Service<HttpRequest, HttpResponse>, BraveService>
-    newDecorator(Tracing tracing) {
-        ensureScopeUsesRequestContext(tracing);
-        return service -> new BraveService(service, tracing);
+    newDecorator(HttpTracing httpTracing) {
+        ensureScopeUsesRequestContext(httpTracing.tracing());
+        return service -> new BraveService(service, httpTracing);
     }
 
     private final Tracer tracer;
     private final TraceContext.Extractor<HttpHeaders> extractor;
+    private final HttpServerHandler<RequestLog, RequestLog> serverHandler;
 
     /**
      * Creates a new instance.
      */
-    private BraveService(Service<HttpRequest, HttpResponse> delegate, Tracing tracing) {
+    private BraveService(Service<HttpRequest, HttpResponse> delegate, HttpTracing httpTracing) {
         super(delegate);
-        tracer = tracing.tracer();
-        extractor = tracing.propagationFactory().create(AsciiStringKeyFactory.INSTANCE)
-                           .extractor(HttpHeaders::get);
+        tracer = httpTracing.tracing().tracer();
+        serverHandler = HttpServerHandler.create(httpTracing, new ArmeriaHttpServerAdapter());
+        extractor = httpTracing.tracing().propagationFactory()
+                               .create(AsciiStringKeyFactory.INSTANCE)
+                               .extractor(HttpHeaders::get);
     }
 
     @Override
     public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
-        final TraceContextOrSamplingFlags contextOrFlags = extractor.extract(req.headers());
-        final Span span = contextOrFlags.context() != null ? tracer.joinSpan(contextOrFlags.context())
-                                                           : tracer.nextSpan(contextOrFlags);
-        // For no-op spans, nothing special to do.
-        if (span.isNoop()) {
-            return delegate().serve(ctx, req);
-        }
-
-        final String method = ctx.method().name();
-        span.kind(Kind.SERVER).name(method);
-        ctx.log().addListener(log -> SpanContextUtil.startSpan(span, log),
```

Comment:
`span.start(log.requestStartTimeMicros())` become `span.start()` in `handleReceive`

Reply:
you can overwrite the start timestamp, but immediately invoking start.timestamp(newOne)

Reply:
I decide to make all annotation using brave's clock, WDYT? @anuraaga .

Reply:
I think we lose timing by using Brave's clock since reasonable time passes from starting the request to calling the decorator (which may be amplified depending on the number and order of decorators). It'd be nice if we can use the more precise Armeria timings IMO.

Start time seems fine to use the overwriting method, and chatting with @adriancole offline he thinks it's reasonable to just copy in the code for `handleFinish` and use our timestamp when finishing since it's a small amount of code. What do you think?

Reply:
+1. We hope Brave adds the variants of `start()` and `finish()` that accept the timestamp given by a caller in the future.

Reply:
not sure am i understand correctly and for confirmation.
We implement our own `handleFinish` and not calling handleSend or handleReceive, right?

Reply:
start timestamp easy, just call span.timestamp later.

what @anuraaga means is copy/paste logic inside handleReceive here for now, just use the external timestamp you have (span.finish(thatTime)) vs implicit

Reply:
updated, so the change is correct?